src/librustc/ty/subst.rs

```diff
@@ -159,10 +159,12 @@ impl<'tcx> Decodable for Kind<'tcx> {
 /// A substitution mapping generic parameters to new values.
 pub type Substs<'tcx> = List<Kind<'tcx>>;
```

Comment:
How do you know you got all cases of `Substs`?

I think one idea would be to hide the `Substs` name somehow - e.g., rename the `pub type Substs<'tcx>` to `pub type InternalSubsts<'tcx>`.